src/openrct2-ui/windows/Footpath.cpp

```diff
@@ -1056,6 +1179,155 @@ namespace OpenRCT2::Ui::Windows
+                    correctedPos.x = _dragStartPos.x;
+                }
+            }
+
+            setMapSelectRange({ _dragStartPos, correctedPos });
+            WindowFootpathSetProvisionalPathCluster(getMapSelectRange(), _footpathPlaceZ);
+        }
+
+        void WindowFootpathPlaceClusterHover(const ScreenCoordsXY& screenCoords)
+        {
+            bool isLeftMousePressed = gInputFlags.has(InputFlag::leftMousePressed);
+            if (isLeftMousePressed)
+                return;
+
+            MapInvalidateSelectionRect();
+            gMapSelectFlags.unset(MapSelectFlag::enableArrow);
+
+            auto mapPos = FootpathGetPlacePositionFromScreenPosition(screenCoords);
+            if (!mapPos)
+            {
+                gMapSelectFlags.unset(MapSelectFlag::enable);
+                return;
+            }
+
+            auto placement = WindowFootpathGetPlacementFromScreenCoords(screenCoords);
+
+            // Set map selection
+            gMapSelectFlags.set(MapSelectFlag::enable);
+            gMapSelectType = MapSelectType::full;
+
+            setMapSelectRange(*mapPos);
+            WindowFootpathSetProvisionalPathCluster(getMapSelectRange(), placement.baseZ);
+        }
+
+        void WindowFootpathPlacePath()
+        {
+            if (_footpathErrorOccured)
+            {
+                return;
+            }
+
+            FootpathUpdateProvisional();
+            if (_provisionalFootpath.tiles.size() == 0)
+                return;
+
+            // Try and place path
+            auto selectedType = gFootpathSelection.GetSelectedSurface();
+            PathConstructFlags constructFlags = FootpathCreateConstructFlags(selectedType);
+
+            bool anySuccess = false;
+            money64 cost = 0;
+            CoordsXYZ lastLocation;
+
+            for (const auto& tile : _provisionalFootpath.tiles)
+            {
+                auto footpathPlaceAction = GameActions::FootpathPlaceAction(
+                    tile.position, tile.slope, selectedType, gFootpathSelection.Railings, kInvalidDirection, constructFlags);
+                footpathPlaceAction.SetCallback(
+                    [&anySuccess, &cost](const GameActions::GameAction* ga, const GameActions::Result* result) {
+                        if (result->Error == GameActions::Status::Ok)
+                        {
+                            anySuccess = true;
+                            cost += result->Cost;
+                        }
+                    });
+                GameActions::Execute(&footpathPlaceAction, getGameState());
+                lastLocation = tile.position;
+            }
+
+            if (anySuccess)
+            {
+                // Don't play sound if it is no cost to prevent multiple sounds. TODO: make this work in no
+                // money scenarios
+                if (cost != 0)
+                {
+                    OpenRCT2::Audio::Play3D(OpenRCT2::Audio::SoundId::placeItem, lastLocation);
```

Comment:
I'm assuming this is based on existing code, but the `OpenRCT2::` specifier should be superfluous here:
```suggestion
                    Audio::Play3D(Audio::SoundId::placeItem, lastLocation);
```